crates/transaction-pool/src/maintain.rs

```diff
@@ -0,0 +1,268 @@
+//! Support for maintaining the state of the transaction pool
+
+use crate::{
+    traits::{CanonicalStateUpdate, ChangedAccount},
+    Pool, TransactionOrdering, TransactionPool, TransactionValidator,
+};
+use futures_util::{Stream, StreamExt};
+use reth_consensus_common::validation::calculate_next_block_base_fee;
+use reth_primitives::{Address, BlockHash, FromRecoveredTransaction};
+use reth_provider::{BlockProvider, CanonStateNotification, PostState, StateProviderFactory};
+use std::{
+    borrow::Borrow,
+    collections::HashSet,
+    hash::{Hash, Hasher},
+};
+use tracing::warn;
+
+/// Maintains the state of the transaction pool by handling new blocks and reorgs.
+///
+/// This listens for any new blocks and reorgs and updates the transaction pool's state accordingly
+pub async fn maintain_transaction_pool<Client, V, T, St>(
```

Comment:
is this gonna be spawned as a separate even handler task or what's the plan here?

Reply:
yes